modules/history/src/views/full/MessageTaskBar.tsx

```diff
@@ -0,0 +1,116 @@
+import { Divider, Icon } from '@blueprintjs/core'
+import { AnchorButton, Colors, Position, Tooltip } from '@blueprintjs/core'
+import { Toolbar } from 'botpress/ui'
+import React, { Fragment } from 'react'
+import { CopyToClipboard } from 'react-copy-to-clipboard'
+
+import { MessageGroup } from '../../typings'
+
+import style from './style.scss'
+import { MessageDownload } from './MessageDownload'
+
+interface Filters {
+  flag: boolean
+}
+
+interface Props {
+  currentConv: string
+  updateFilters: (f: Filters) => void
+  useAsFilter: boolean
+  selectedCount: number
+  flag: () => void
+  unflag: () => void
+  messageGroups: MessageGroup[]
+}
+
+interface State {
+  filters: Filters
+  currentConv: string
+  showCopiedTooltip: boolean
+}
+
+export class MessageTaskBar extends React.Component<Props, State> {
+  state = {
+    filters: {
+      flag: false
+    },
```

Comment:
Please avoid objects in the state, it's way easier to use filterFlags: false, then create your object if you need it when creating the query. Setting state objects isn't very clear

Reply:
Yeah I know, it is only old code that I have not refactored yet because it is already working. It's not new logic, it has only been moved to another file. I can make the refactor, but it doesn't really add any value to the product.